### PR TITLE
Support converting cipher name to RFC name and vice versa

### PR DESCRIPTION
**Pull request by InfoHunter**
Fixes: issue #3747

make SSL_CIPHER_standard_name globally available and introduce a new
function SSL_CIPHER_ossl_name.

Documentation and test cases are added.

A new option `-convert` is also added to `openssl ciphers` app to provide a convenient use from command line to convert 'standard name' to 'OpenSSL name'.

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by InfoHunter** [comment]
> Is there a concern about ideology?

Don't fully understand 'ideology'...

**Comment by levitte** [comment]
I'll say this to you as I said it to @paulidale; in the future, please make coding style fixups a separate commit, to make it easier to review them separately from the more substantial changes.

**Comment by InfoHunter** [comment]
> I'll say this to you as I said it to @paulidale; in the future, please make coding style fixups a separate commit, to make it easier to review them separately from the more substantial changes.

Copy that.

**Comment by richsalz** [comment]
By ideology, he means are we bothered that people will use standard terms and not OpenSSL's terms for crypto.  Do we want to "force" people to use our words?

The answer is no. :)

**Comment by richsalz** [comment]
Should there be an option to ‘openssl ciphers’ that prints the standard name?

**Comment by InfoHunter** [comment]
> Should there be an option to ‘openssl ciphers’ that prints the standard name?

There is one, `-stdname`. But this option was available only if 'SSL trace' feature is enabled in the past. This time I made `-stdname` globally available either.

**Comment by InfoHunter** [comment]
> By ideology, he means are we bothered that people will use standard terms and not OpenSSL's terms for crypto.

Interesting, where did the OpenSSL's name come from historically?

**Comment by richsalz** [comment]
I don’t know where the original names came from.  Maybe @t-j-h recalls.

**Comment by richsalz** [comment]
As for -stdname, nice work.  I should have re-read the PR before commenting :)

File: include/openssl/ssl.h [hunk 1]
```diff
@@ -1416,6 +1416,8 @@ __owur const SSL_CIPHER *SSL_get_current_cipher(const SSL *s);
 __owur int SSL_CIPHER_get_bits(const SSL_CIPHER *c, int *alg_bits);
 __owur const char *SSL_CIPHER_get_version(const SSL_CIPHER *c);
 __owur const char *SSL_CIPHER_get_name(const SSL_CIPHER *c);
+__owur const char *SSL_CIPHER_standard_name(const SSL_CIPHER *c);
+__owur const char *SSL_CIPHER_ossl_name(const char *rfc_name);
```
Comment (tmshort) [inline include/openssl/ssl.h]: The SSL_CIPHER prefix usually implies that the first argument is an SSL_CIPHER object; which is not the case here. Is there some "better" name that could be used instead?

Comment (InfoHunter) [inline include/openssl/ssl.h]: It might be, I have considered this either, but couldn't find a proper name. Any other suggest?

**Comment by richsalz** [comment]
How about “const char *OPENSSL_cipher_name(const char *)”

**Comment by InfoHunter** [comment]
> How about “const char *OPENSSL_cipher_name(const char *)”

Sounds better : )

**Comment by InfoHunter** [comment]
Change `SSL_CIPHER_ossl_name` to `OPENSSL_cipher_name`.
Is it 'neat' to put this `OPENSSL_cipher_name` among a bundle of `SSL_CIPHER_*` stuffs, e.g. at function declarations or in documentation?

**Comment by richsalz** [comment]
sure, put it with the other ssl-cipher names

**Comment by InfoHunter** [comment]
> sure, put it with the other ssl-cipher names

Yep,  seems everything's fine now. @tmshort

**Comment by InfoHunter** [comment]
Ping?

**Comment by mattcaswell** [comment]
Ping @levitte or @paulidale. Looks like this PR has the necessary approvals. Will one of you push?

**Comment by mattcaswell** [comment]
Ahh...actually there was an update after your approvals, so you probably need to reconfirm.

**Comment by InfoHunter** [comment]
pushed to solve conflicts

**Comment by InfoHunter** [comment]
Any other comments for this PR? @levitte @paulidale ?

**Comment by InfoHunter** [comment]
Pushed to update copyright information, as a part of PR #3950

**Comment by mattcaswell** [comment]
I have no objection to the copyrights in this PR. Basically you have added BaishanCloud copyrights where the contribution constitutes the entirety of the file - which seems reasonable.

**Comment by richsalz** [comment]
@levitte @paulidale please reconfirm if you don't object to the copyright.

**Comment by InfoHunter** [comment]
Should I push a new commit to resolve the conflicts or update the original commit? (based on #3963 )

**Comment by InfoHunter** [comment]
I decided to resolve this conflict by 'rebase' and 'push -f' this time, and wait for an agreement in #3963 .

**Comment by InfoHunter** [comment]
please re-confirm this update @levitte @paulidale , only `test/build.info` file is changed to fix conflicts.

**Comment by paulidale** [comment]
Pushed.

**Pull request closed without merging**